### PR TITLE
Refactor Backups: reuse common code between Azure and GCS

### PR DESCRIPTION
**Pull request by rodrigo-lourenco-lopes**
## Description

- Move manifests implementation, file sets, and some exceptions to backup.common

So far these common parts of the code were only being repeated between Azure and GCS, some further changes would be necessary to also abstract from S3. Additionally address some raised issues by the commit checker.

This will not close the original issue, as there seem to be other possible refactors to be done.

## Related issues

relates https://github.com/camunda/zeebe/issues/16105

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Other teams:
If the change impacts another team an issue has been created for this team, explaining what they need to do to support this change.
- [ ] [Operate](https://github.com/camunda/operate/issues)
- [ ] [Tasklist](https://github.com/camunda/tasklist/issues)
- [ ] [Web Modeler](https://github.com/camunda/web-modeler/issues)
- [ ] [Desktop Modeler](https://github.com/camunda/camunda-modeler/issues)
- [ ] [Optimize](https://github.com/camunda/camunda-optimize/issues)

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

**Review by lenaschoenburg — CHANGES_REQUESTED** [review]
:x: I don't think these types should be in the backup module, I'd rather have a new module in `backup-stores/common`.

**Comment by rodrigo-lourenco-lopes** [comment]
> ❌ I don't think these types should be in the backup module, I'd rather have a new module in `backup-stores/common`.

@oleschoenburg if we abstract these types to a new module we get a circular reference between this new module and the backup, unless we move the method toStaus() back to the respective `AzureBackupStore` and `GcsBackupStore` respectively. Is it worth it?

**Comment by oleschoenburg** [comment]
Could we just move the `toStatus` method somewhere else? AFAICT it doesn't need to be in the `backup` module.

File: zeebe/backup-stores/common/pom.xml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  <dependencies>
+    <dependency>
+      <groupId>io.camunda</groupId>
+      <artifactId>zeebe-backup</artifactId>
+    </dependency>
+
+    <dependency>
+      <groupId>com.fasterxml.jackson.core</groupId>
+      <artifactId>jackson-databind</artifactId>
+      <version>2.16.1</version>
```
Comment (oleschoenburg) [inline zeebe/backup-stores/common/pom.xml:37]: :x: We should not specify a version here, we should use the one from the parent.

**Review by lenaschoenburg — CHANGES_REQUESTED** [review]

File: zeebe/backup-stores/common/pom.xml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+  <artifactId>zeebe-backup-store-common</artifactId>
+  <packaging>jar</packaging>
+
+  <name>Zeebe Backup Store Common</name>
+
+  <properties>
+    <maven.compiler.source>21</maven.compiler.source>
+    <maven.compiler.target>21</maven.compiler.target>
+    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
```
[the remark this answers is not in the document]
Comment (oleschoenburg) [reply zeebe/backup-stores/common/pom.xml:26]: I reopened because it's not resolved yet.

**Review by lenaschoenburg — APPROVED** [review]
Thanks @rodrigo-lourenco-lopes :+1: 

I approved now but please address the one remaining issue. I think you accidentally marked it as resolved before.

File: zeebe/backup-stores/common/pom.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (rodrigo-lourenco-lopes) [reply zeebe/backup-stores/common/pom.xml:26]: ups, yes, I think i resolved it accidentally

**Pull request merged**